packages/core/src/repositories/documentVersionsRepository.ts

```diff
@@ -0,0 +1,211 @@
+import { NotFoundError, Result } from '$core/lib'
+import {
+  Commit,
+  commits,
+  DocumentVersion,
+  documentVersions,
+  projects,
+} from '$core/schema'
+import { and, eq, getTableColumns, isNotNull, lte, max } from 'drizzle-orm'
+
+import Repository from './repository'
+
+export type GetDocumentAtCommitProps = {
+  commit: Commit
+  documentUuid: string
+}
+
+export class DocumentVersionsRepository extends Repository {
+  get scope() {
+    return this.db
+      .select(getTableColumns(documentVersions))
+      .from(documentVersions)
+      .innerJoin(projects, eq(projects.workspaceId, this.workspaceId))
+      .innerJoin(commits, eq(commits.projectId, projects.id))
+      .where(eq(documentVersions.commitId, commits.id))
+      .as('documentVersionsScope')
+  }
+
+  async getDocumentById(documentId: number) {
+    const res = await this.db
+      .select()
+      .from(this.scope)
+      .where(eq(this.scope.id, documentId))
+
+    // NOTE: I hate this
+    const document = res[0]
```

Comment:
```javascript
this.db.documents.findFirst({
  where: eq(this.scope.id, documentId)
})
```

Reply:
nein, that's one of the negative effects of using subqueries you cannot use drizzle query dsl